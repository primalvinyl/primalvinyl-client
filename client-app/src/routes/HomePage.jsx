import React from 'react';
import MasterTemplate from '../components/MasterTemplate';
import './HomePage.scss';

const HomePage = () => {
    return (
        <MasterTemplate>
            <div className="home-page">
                <div className="wrapper">
                    <h2>Home Page</h2>
                </div>
            </div>
        </MasterTemplate>
    );
};

export default HomePage;
