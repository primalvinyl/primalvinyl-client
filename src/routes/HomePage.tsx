import React from 'react';
import MasterTemplate from '../components/_MasterTemplate';
import SearchForm from '../components/presentation/SearchForm';
import styles from './HomePage.module.scss';

const HomePage = (): React.ReactElement => {
    return (
        <MasterTemplate>
            <article className={styles.root}>
                <div className={styles.wrapper}>
                    <h2>Home Page</h2>
                    <SearchForm />
                </div>
            </article>
        </MasterTemplate>
    );
};

export default HomePage;
