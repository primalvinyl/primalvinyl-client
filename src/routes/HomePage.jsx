import React from 'react';
import { useDispatch } from 'react-redux';
import { getArtist } from '../actions';
import MasterTemplate from '../components/MasterTemplate';
import './HomePage.scss';

const HomePage = () => {
    const dispatch = useDispatch();

    const handleSubmit = event => {
        dispatch(getArtist('278763'));
    };
    return (
        <MasterTemplate>
            <article className="home-page">
                <div className="wrapper">
                    <h2>Home Page</h2>
                    <div>
                        <input type="text" />
                        <button onClick={handleSubmit}>
                            Search
                        </button>
                    </div>
                </div>
            </article>
        </MasterTemplate>
    );
};

export default HomePage;
