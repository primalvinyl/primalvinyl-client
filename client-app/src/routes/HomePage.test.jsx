import React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import HomePage from './HomePage';

describe('HomePage', () => {
    it('renders', () => {
        const myWrapper = shallow(<HomePage />);
        expect(myWrapper.exists());
    });
    it('matches snapshot', () => {
        const component = Renderer.create(<HomePage />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
