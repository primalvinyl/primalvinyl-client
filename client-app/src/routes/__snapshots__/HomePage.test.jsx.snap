// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage matches snapshot 1`] = `
<div
  className="home-page"
>
  <div
    className="wrapper"
  >
    <h2>
      Home Page
    </h2>
  </div>
</div>
`;
